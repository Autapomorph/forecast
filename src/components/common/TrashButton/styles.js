import styled from 'styled-components';

// eslint-disable-next-line import/prefer-default-export
export const StyledTrashButton = styled.span`
  color: var(--red);
  border: 0;
  cursor: pointer;

  &:hover {
    color: var(--dark-red);
  }
`;
