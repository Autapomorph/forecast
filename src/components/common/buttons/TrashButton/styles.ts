import styled from 'styled-components/macro';

import BaseIconButton from 'components/common/buttons/BaseIconButton';

export const TrashButton = styled(BaseIconButton)`
  color: var(--red);

  :hover {
    color: var(--dark-red);
  }
`;
