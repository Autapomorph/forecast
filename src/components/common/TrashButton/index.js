import React, { Fragment } from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import faTrashAlt from '@fortawesome/fontawesome-free-solid/faTrashAlt';

import { StyledTrashButton } from './styles';

const TrashButton = ({ isEmpty = false, onClick = () => {} }) => (
  <Fragment>
    {!isEmpty && (
      <StyledTrashButton onClick={onClick}>
        <FontAwesomeIcon icon={faTrashAlt} size="lg" />
      </StyledTrashButton>
    )}
  </Fragment>
);

export default TrashButton;
