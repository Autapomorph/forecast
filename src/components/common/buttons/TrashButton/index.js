import React, { Fragment } from 'react';
import { faTrashAlt } from '@fortawesome/free-solid-svg-icons';

import { StyledTrashButton } from './styles';

const TrashButton = ({ isEmpty = false, onClick = () => {} }) => (
  <Fragment>
    {!isEmpty && <StyledTrashButton icon={faTrashAlt} onClick={onClick} size="lg" />}
  </Fragment>
);

export default TrashButton;
