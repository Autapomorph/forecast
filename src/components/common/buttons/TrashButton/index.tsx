import React from 'react';
import { faTrashAlt } from '@fortawesome/free-solid-svg-icons';
import { SizeProp } from '@fortawesome/fontawesome-svg-core';

import * as S from './styles';

type Props = {
  isEmpty: boolean;
  onClick?: () => void;
  size?: SizeProp;
};

const TrashButton: React.FC<Props> = ({
  isEmpty = false,
  size,
  onClick,
  ...props
}): React.ReactElement => (
  <>{!isEmpty && <S.TrashButton icon={faTrashAlt} onClick={onClick} size={size} {...props} />}</>
);

export default TrashButton;
