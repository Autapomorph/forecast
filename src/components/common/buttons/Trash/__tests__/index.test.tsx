import React from 'react';
import { shallow, mount } from 'enzyme';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

import TrashButton from '..';

describe('TrashButton tests', () => {
  it('renders properly', () => {
    const wrapper = shallow(<TrashButton isEmpty={false} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render FontAwesomeIcon if not empty', () => {
    const wrapper = mount(<TrashButton isEmpty={false} />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(FontAwesomeIcon)).toHaveLength(1);
  });

  it('should not render FontAwesomeIcon if empty', () => {
    const wrapper = mount(<TrashButton isEmpty />);

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(FontAwesomeIcon)).toHaveLength(0);
  });

  it('should handle clicks', () => {
    const onClick = jest.fn();
    const wrapper = mount(<TrashButton isEmpty={false} onClick={onClick} />);
    wrapper.find(FontAwesomeIcon).simulate('click');

    expect(onClick).toHaveBeenCalled();
  });
});
