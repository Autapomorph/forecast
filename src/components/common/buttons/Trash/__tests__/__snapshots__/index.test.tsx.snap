// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrashButton tests renders properly 1`] = `
<Fragment>
  <styles__TrashButton
    icon={
      Object {
        "icon": Array [
          448,
          512,
          Array [],
          "f2ed",
          "M32 464a48 48 0 0 0 48 48h288a48 48 0 0 0 48-48V128H32zm272-256a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zM432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16z",
        ],
        "iconName": "trash-alt",
        "prefix": "fas",
      }
    }
  />
</Fragment>
`;

exports[`TrashButton tests should not render FontAwesomeIcon if empty 1`] = `
<TrashButton
  isEmpty={true}
/>
`;

exports[`TrashButton tests should render FontAwesomeIcon if not empty 1`] = `
.c1 {
  padding: 5px;
  border: none;
  background: none;
  color: inherit;
  cursor: pointer;
}

.c1:hover {
  color: var(--white-dark);
}

.c0 {
  color: var(--red);
}

.c0:hover {
  color: var(--red-dark);
}

<TrashButton
  isEmpty={false}
>
  <styles__TrashButton
    icon={
      Object {
        "icon": Array [
          448,
          512,
          Array [],
          "f2ed",
          "M32 464a48 48 0 0 0 48 48h288a48 48 0 0 0 48-48V128H32zm272-256a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zM432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16z",
        ],
        "iconName": "trash-alt",
        "prefix": "fas",
      }
    }
  >
    <BaseIconButton
      className="c0"
      icon={
        Object {
          "icon": Array [
            448,
            512,
            Array [],
            "f2ed",
            "M32 464a48 48 0 0 0 48 48h288a48 48 0 0 0 48-48V128H32zm272-256a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zM432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16z",
          ],
          "iconName": "trash-alt",
          "prefix": "fas",
        }
      }
    >
      <styles__IconButton
        className="c0"
        onClick={[Function]}
      >
        <button
          className="c1 c0"
          onClick={[Function]}
          type="button"
        >
          <FontAwesomeIcon
            border={false}
            className=""
            fixedWidth={false}
            flip={null}
            icon={
              Object {
                "icon": Array [
                  448,
                  512,
                  Array [],
                  "f2ed",
                  "M32 464a48 48 0 0 0 48 48h288a48 48 0 0 0 48-48V128H32zm272-256a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zM432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16z",
                ],
                "iconName": "trash-alt",
                "prefix": "fas",
              }
            }
            inverse={false}
            listItem={false}
            mask={null}
            pull={null}
            pulse={false}
            rotation={null}
            size="lg"
            spin={false}
            swapOpacity={false}
            symbol={false}
            title=""
            transform={null}
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-trash-alt fa-w-14 fa-lg "
              data-icon="trash-alt"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M32 464a48 48 0 0 0 48 48h288a48 48 0 0 0 48-48V128H32zm272-256a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zM432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </FontAwesomeIcon>
        </button>
      </styles__IconButton>
    </BaseIconButton>
  </styles__TrashButton>
</TrashButton>
`;
